Documentation/Proposals/NNNN-exit-tests.md

```diff
@@ -0,0 +1,421 @@
+# Exit tests
+
+* Proposal: [SWT-NNNN](NNNN-exit-tests.md)
+* Authors: [Jonathan Grynspan](https://github.com/grynspan)
+* Status: **Awaiting review**
+* Bug: [apple/swift-testing#157](https://github.com/apple/swift-testing/issues/157)
+* Implementation: [apple/swift-testing#307](https://github.com/apple/swift-testing/pull/307)
+* Review: TBD <!-- ([pitch](https://forums.swift.org/...)) -->
+
+## Introduction
+
+One of the first enhancement requests we received for swift-testing was the
+ability to test for precondition failures and other critical failures that
+terminate the current process when they occur. This feature is also frequently
+requested for XCTest. With swift-testing, we have the opportunity to build such
+a feature in an ergonomic way.
+
+> [!NOTE]
+> This feature has various names in the relevant literature, e.g. "exit tests",
+> "death tests", "death assertions", etc. We consistently use the term "exit
+> tests" to refer to them.
```

Comment:
"Termination tests" is another term I have heard for this concept which may be worth mentioning